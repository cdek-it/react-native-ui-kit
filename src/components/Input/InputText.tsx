import { memo } from 'react'
import { View, type ViewStyle } from 'react-native'

import { InputTextBase } from './InputTextBase/InputTextBase'
import type { InputTextBaseProps } from './InputTextBase/types'

export interface InputTextProps extends InputTextBaseProps {
  /** Дополнительная стилизация для контейнера компонента */
  style?: ViewStyle
}

/**
 * Компонент для ввода текста
 * @link https://www.figma.com/design/4TYeki0MDLhfPGJstbIicf/UI-kit-PrimeFace-(DS)?node-id=484-5470&m=dev
 * @see InputTextBase
 * @see InputTextBaseProps
 */
export const InputText = memo<InputTextProps>(({ style, ...otherProps }) => {
  return (
    <View style={style}>
      <InputTextBase {...otherProps} />
    </View>
  )
})
