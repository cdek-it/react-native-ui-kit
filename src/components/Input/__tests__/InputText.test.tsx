import { render } from '@testing-library/react-native'

import { InputText } from '../InputText'

jest.mock('../InputTextBase/InputTextBase', () => ({
  InputTextBase: 'InputTextBase',
}))

describe('InputText component tests', () => {
  test('snapshot', () => {
    const { toJSON } = render(<InputText style={{ margin: 10 }} value='text' />)

    expect(toJSON()).toMatchSnapshot()
  })
})
