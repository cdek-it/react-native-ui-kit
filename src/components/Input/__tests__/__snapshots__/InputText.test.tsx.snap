// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText component tests snapshot 1`] = `
<View
  style={
    {
      "margin": 10,
    }
  }
>
  <InputTextBase
    value="text"
  />
</View>
`;
